.github/actions/next-repo-actions/lib/util.mjs

```diff
@@ -15,3 +18,47 @@ export function ninetyDaysAgo() {
   date.setDate(date.getDate() - 90)
   return date.toISOString().split('T')[0]
 }
+
+export async function getLatestCanaryVersion() {
+  let latestCanaryVersion
+
+  try {
+    const { stdout } = await getExecOutput('pnpm', [
+      'view',
+      'next',
+      'dist-tags',
+      '--json',
+    ])
+
+    const distTags = JSON.parse(stdout)
+    latestCanaryVersion = distTags.canary || null
+  } catch (error) {
+    core.error(`Error fetching latest Next.js canary version, skipping update.`)
+
+    core.exit(core.ExitCode.Failure)
+  }
+
+  return latestCanaryVersion
+}
+
+export async function getLatestVersion() {
+  let latestVersion
+
+  try {
+    const { stdout } = await getExecOutput('pnpm', [
+      'view',
+      'next',
+      'dist-tags',
+      '--json',
+    ])
+
+    const distTags = JSON.parse(stdout)
+    latestVersion = distTags.latest || null
+  } catch (error) {
+    core.error(`Error fetching latest Next.js version, skipping update.`)
+
+    core.exit(core.ExitCode.Failure)
```

Comment:
We shouldn't swallow the error here. This makes debugging flakiness impossible since we may never see the error when we look for it. In other places we use `setFailed`. Can we use `setFailed` here as well?

Reply:
Noted, will update! <picture data-single-emoji=":frog-thumbsup:" title=":frog-thumbsup:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/frog-thumbsup/74768cbdae647b4d.png" alt=":frog-thumbsup:" align="absmiddle"></picture>